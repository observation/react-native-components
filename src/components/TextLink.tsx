import React from 'react'
import { StyleProp, Text, TextStyle, TouchableOpacity, ViewStyle } from 'react-native'

import appTextStyle from '../styles/text'

type Props = {
  text: string
  onPress?: () => void
  containerStyle?: StyleProp<ViewStyle>
  textStyle?: StyleProp<TextStyle>
}

const TextLink = ({ text, onPress, containerStyle, textStyle }: Props) => (
  <TouchableOpacity style={containerStyle} onPress={onPress} activeOpacity={0.5}>
    <Text style={[appTextStyle.link, textStyle]}>{text}</Text>
  </TouchableOpacity>
)

export default TextLink
