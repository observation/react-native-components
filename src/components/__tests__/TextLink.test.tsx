import React from 'react'

import { render, fireEvent } from '@testing-library/react-native'

import TextLink from '../TextLink'

const onPress = jest.fn()
describe('TextLink', () => {
  test('Rendering enabled', () => {
    const { toJSON } = render(
      <TextLink text="Text to click" onPress={onPress} containerStyle={{ flex: 1 }} textStyle={{ fontSize: 1 }} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  test('Rendering disabled', () => {
    const { toJSON } = render(
      <TextLink text="Text to click" onPress={onPress} containerStyle={{ flex: 1 }} textStyle={{ fontSize: 1 }} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  test('Click', async () => {
    const { getByText } = render(
      <TextLink text="Text to click" onPress={onPress} containerStyle={{ flex: 1 }} textStyle={{ fontSize: 1 }} />,
    )
    await fireEvent.press(getByText('Text to click'))
    expect(onPress).toBeCalled()
  })
})
