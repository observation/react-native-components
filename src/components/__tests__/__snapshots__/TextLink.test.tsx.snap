// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink Rendering disabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flex": 1,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#0066B1",
          "fontFamily": "Ubuntu",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": 20,
        },
        {
          "fontSize": 1,
        },
      ]
    }
  >
    Text to click
  </Text>
</View>
`;

exports[`TextLink Rendering enabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flex": 1,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#0066B1",
          "fontFamily": "Ubuntu",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": 20,
        },
        {
          "fontSize": 1,
        },
      ]
    }
  >
    Text to click
  </Text>
</View>
`;
